src/IconWithLabel/IconWithLabel.test.tsx

```diff
@@ -0,0 +1,39 @@
+import { describe, expect, test } from "vitest";
+import { render, screen } from "@testing-library/react";
+
+import { CarMakerLogo } from "../SvgIcons";
+import IconWithLabel from "./IconWithLabel";
+import React from "react";
+
+const defaultInputs = {
+  icon: <CarMakerLogo />,
+  label: "Example"
+};
+
+describe("IconWithLabel tests", () => {
+  test("renders `IconWithLabel` with a href", () => {
+    render(<IconWithLabel {...defaultInputs} href={"https://example.com"} />);
+
+    // find the elements of interest
+    const iconWithLabelElement = screen.getByTestId("icon-with-label");
+    const anchorElement = screen.getByRole("link");
+
+    // check if elements captured match expectations
+    expect(iconWithLabelElement).toBeInTheDocument();
+    expect(iconWithLabelElement).toHaveTextContent("Example");
+    expect(anchorElement).toBeInTheDocument();
+    expect(anchorElement).toHaveAttribute("href", "https://example.com");
+  });
+  test("renders `IconWithLabel` without an href", () => {
+    render(<IconWithLabel {...defaultInputs} />);
+
+    // find the elements of interest
+    const iconWithLabelElement = screen.getByTestId("icon-with-label");
+    const anchorElement = screen.queryByRole("link");
+
+    // check if elements captured match expectations
+    expect(iconWithLabelElement).toBeInTheDocument();
+    expect(iconWithLabelElement).toHaveTextContent("Example");
+    expect(anchorElement).not.toBeInTheDocument();
+  });
```

Comment:
Shouldn't we also have a test case for truncating?

Reply:
There is no way to test exactly if the text has truncated in the react-testing-library, as it doesn't render this component visually. However, we can check that the text field has all the necessary css props so that we are sure that it "should" be truncating. I will implement this.